src/Handler/CurlFactory.php

```diff
@@ -338,6 +348,7 @@ private function applyHeaders(EasyHandle $easy, array &$conf): void
      */
     private function removeHeader(string $name, array &$options): void
     {
+        /** @var string $key */
```

Comment:
This is not ideal, but requires `$options` to be an array-shape, as there is no sanity check that the `_headers` index actually exists here which would also allow better type inference